plugins/3MFReader/ThreeMFWorkspaceReader.py

```diff
@@ -1084,10 +1082,7 @@ def read(self, file_name):
             CuraApplication.getInstance().getMachineManager().activeQualityChanged.emit()
 
         # Actually change the active machine.
-        Application.getInstance().setGlobalContainerStack(global_stack)
-
-        # Notify everything/one that is to notify about changes.
-        global_stack.containersChanged.emit(global_stack.getTop())
+        CuraApplication.getInstance().callLater(self._updateActiveMachine, global_stack)
```

Comment:
Why is this updated from the Qt thread?

Reply:
[will doc this]

Now the GUI and `MachineManager` depends on the new lookup managers we made here. Those managers will update their lookup trees whenever they get a container/container metadata changed signal.

In project loading, `read()` may modify/create containers so that will trigger an update on the lookup tress later. However, `read()` is executed on the Qt thread, which means all emitted signal will not take effect immediately. Because `_updateActiveMachine()` depends on the lookup tables to have latest data, if we call it directly here, it will only get the current dataset. So, to solve this, we delay this call by schedule it on the Qt thread for later.

Reply:
Added doc in ca88e0df8

Reply:
Aside from documentation, this creates a dependency in which order the Qt events should be handled. Yes, it is FIFO, but it's difficult to maintain an application that depends on this order.

Perhaps we can instead let `_updateActiveMachine` listen to a different signal that gets fired after the `read()` function for project loading is done?